plugins/parsers/value/parser.go

```diff
@@ -23,6 +23,7 @@ func (v *ValueParser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	if v.DataType == "string" {
 		vStr = strings.TrimSpace(string(buf))
 	} else {
+		buf = bytes.Trim(buf, "\x00")
```

Comment:
yes, please do that, in that case the first step of the function should be to do `buf = bytes.TrimSpace(bytes.Trim(buf, "\x00"))`